### PR TITLE
Implement probability map smoothing via convolutions

### PR DESCRIPTION
**Pull request by dusenberrymw**
This implements probability map smoothing via a convolution with a disk
kernel as a means of reducing the noise for stride 1, 0 probability
threshold probability predictions.  Additionally, this adds a mitosis
detection algorithm based on related literature.  Early experiments show
promising results.

### CODE REVIEW

**Comment by dusenberrymw** [comment]
@feihugis Can you please review this?  I decided to make a new module so that we could keep the f1 evaluation separate from the detection stuff.  It might be nice to move the clustering code here as well.

**Comment by dusenberrymw** [comment]
Also, there is some inefficiency with how the `conv_smooth` function creates a new tf session during every call.  I hate bleeding the tensorflow graph execution out into the code base, as it makes it hard for someone to use the function, but it makes it less efficient.  I might explore having the `conv_smooth` function actually return a lambda closure over the graph and `sess.run` junk so that someone can just get the lambda and call it with a numpy input, without incurring the cost of the session creation or large graph creation, and without having to directly deal with the graph.

File: breastcancer/detection.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+def ijv_2_arr(ijv, h, w):
+  arr = np.zeros((h, w), dtype=np.float32)
+  arr[ijv[:,0].astype(np.int), ijv[:,1].astype(np.int)] = ijv[:,2]
+  return arr
+
+
+def arr_2_ijv(arr):
+  r, c = np.nonzero(arr >= 0)
+  ijv = np.vstack((r, c, arr[r,c])).T  # row, col, prob
+  return ijv
```
Comment (feihugis) [inline breastcancer/detection.py]: Could you add some comments for the two function?

**Comment by dusenberrymw** [comment]
@feihugis I addressed your great comments.  Can you take one more look?

**Comment by feihugis** [comment]
@dusenberrymw Thanks! LGTM!

**Pull request closed without merging**